### ISSUE TITLE
Webhook skipped due to invalid custom data.

### ISSUE THREAD

**Issue by VitalijOd** [lmsqueezy/laravel#57]
Hello! I'm catching an error 'Webhook skipped due to invalid custom data' for the 'order_created' and 'subscription_created' webhooks.

According to the source code, the webhook should contain 'billable_id' and 'billable_type' in the custom data, but I'm receiving webhooks without these fields.

Do I need to pass these fields manually? Sorry to bother you, but I'm struggling to find any info on this.

**Comment by driesvints** [comment]
Hi there. Can you provide some code to reproduce this issue?

**Comment by atmonshi** [comment]
I am getting the same thing `Webhook skipped due to invalid custom data`
with no exceptions or any other logs

would it be a good idea if the `WebhookController` reports the exception when caught, so we get better tracing?

```php
try {
    $this->{$method}($payload);
} catch (InvalidCustomPayload $e) {
    report($e);
    return new Response('Webhook skipped due to invalid custom data.');
}
```

thank you for the package btw :)

**Comment by VitalijOd** [comment]
> Hi there. Can you provide some code to reproduce this issue?

No custom code is required. I've simply set up the webhooks to target the '/lemon-squeezy/webhook' route.

In the package WebhookController.php, the handleOrderCreated method calls the 'resolveBillable' method, which requires the attribute $payload['meta']['custom_data']['billable_id']. However, this attribute is missing in the webhook payload, resulting in the 'InvalidCustomPayload' error being thrown.
<img width="1454" alt="SCR-20231122-prlp" src="https://github.com/lmsqueezy/laravel/assets/78653500/6fb7ee9d-0dc5-4757-b72f-ad8b71a76ec5">

**Comment by driesvints** [comment]
@VitalijOd but that's what I mean. This package requires you to use the code of the package to start subscriptions and perform orders. If you don't do that then those custom data keys aren't present in your payloads. 

So the right question might be: how are your customers buying your products? Is there code involved? And if not, how are you serving them the checkout page?

**Comment by driesvints** [comment]
@atmonshi heya thanks!

This exception will go away once https://github.com/lmsqueezy/laravel/issues/53 is implemented so hopefully that's not much longer.

**Comment by VitalijOd** [comment]
> @VitalijOd but that's what I mean. This package requires you to use the code of the package to start subscriptions and perform orders. If you don't do that then those custom data keys aren't present in your payloads.
> 
> So the right question might be: how are your customers buying your products? Is there code involved? And if not, how are you serving them the checkout page?

Thank you for the clarification. I generated the payment link without this package and intended to use it solely for handling webhooks. I now understand this is likely why the webhook payload didn't contain billable_id. I think #53 will resolve my problem in this case.

**Comment by driesvints** [comment]
Right, yeah right now that won't work. You need to use the package itself in order to make the webhooks work.

**Comment by calebporzio** [comment]
So this has been baffling me as well and I think I found my issue:
* Orders created from "recovered cart" emails sent by LS by default.

I disabled those emails and we'll see if it fixes it. However, it would be nice to have the cart recovery features, so hoping #53 gets built as well!

Just came here to provide more context on a solution for people.

I also disabled my storefront that LS creates by default. That was another area that could have been leaking purchases.

**Comment by driesvints** [comment]
Thanks for that feedback @calebporzio. I'll try to see if I can get #53 in soon.
